### PR TITLE
feat: add `aws-vault` spec

### PR DESCRIPTION
**Pull request by swallowtail62**
This PR adds [aws-vault](https://github.com/99designs/aws-vault) completion spec.

### CODE REVIEW

**Comment by swallowtail62** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/aws-vault.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
+const profilesGeneratro: Fig.Generator = {
```
Comment (fedeci) [inline src/aws-vault.ts:1]: typo:
```suggestion
const profilesGenerator: Fig.Generator = {
```

File: src/aws-vault.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  name: "aws-vault",
+  description:
+    "A vault for securely storing and accessing AWS credentials in development environments",
+  subcommands: [
+    {
+      name: "help",
+      description: "Show help about the command",
+      args: {
+        name: "command",
+        suggestions: helpSuggestions,
```
Comment (fedeci) [inline src/aws-vault.ts:146]: This is a nice sugar we provide :)
```suggestion
        template: "help",
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks @swallowtail62! I left a couple minor issues to solve but the spec looks good overall.

File: src/aws-vault.ts (hunk 1 above)
Comment (swallowtail62) [reply src/aws-vault.ts]: Oops, sorry. Thanks for pointing that out!

File: src/aws-vault.ts (hunk 2 above)
Comment (swallowtail62) [reply src/aws-vault.ts]: @fedeci 
Thanks! I fixed it, check it out.

**Review by fedeci — APPROVED** [review]

**Comment by fedeci** [comment]
Thanks @swallowtail62!

**Pull request merged**